scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -0,0 +1,515 @@
+
+
+```{r merge sces}
+merged_sce <- do.call(cbind, sce_list_formatted)
+merged_sce
+```
+We now have a single SCE object that contains all cells from all libraries we'd like to integrate.
+
+## Integration
+
+### First, explore! (needs better title?)
+
+You might have noticed that that there are already `PCA` and `UMAP` dimension reductions in the merged SCE we created.
+These represent the original dimension reductions that were performed on _each individual library_.
+
+We can also calculate PCA and UMAP for the merged SCE as a whole, which we will save with different names `merged_PCA` and `merged_UMAP` to distinguish them for the original.
+When calculating PCs on this merged object, we need to use a slightly different approach that respects the batch differences within. 
+We can compare these UMAP embeddings to those calculated after performing integration.
+
+
+```{r}
+# calculate high-variance genes to use for dimension reduction
+gene_variance <- scran::modelGeneVar(merged_sce,
+                                     # specify the grouping/batch column:
+                                     block = merged_sce$sample)
+hv_list <- scran::getTopHVGs(gene_variance,
+                              n = 2000)
+
+
+# Use batchelor to calculate PCA from a merged group fo SCEs
+multi_pca <- batchelor::multiBatchPCA(merged_sce,
+                                      subset.row = hv_list,
+                                      batch = merged_sce$sample,
+                                      preserve.single = TRUE)
+
+# add PCA results to merged SCE object 
+reducedDim(merged_sce, "merged_PCA") <- multi_pca@listData[[1]]
+
+# add merged_UMAP from merged_PCA
+merged_sce <- scater::runUMAP(merged_sce,
+                              dimred = "merged_PCA",
+                              name = "merged_UMAP")
+
+merged_sce
+```
+
+
+Let's first see how these uncorrected UMAPs compare:
+
+```{r uncorrected UMAPs}
+# UMAPs scaled separately when calculated from individual libraries:
+scater::plotReducedDim(merged_sce,
+                       dimred = "UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) +
+  ggplot2::ggtitle("UMAPs separately calculated")
+
+# UMAPs scaled together when calculated from the merged SCE
+scater::plotReducedDim(merged_sce,
+                       dimred = "merged_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) +
+  ggplot2::ggtitle("UMAPs calculated on merged data")
+```
+The main "clustering" (not technical clustering, just what we see visually!) in these plots are the batches. 
+A well-integrated dataset will instead show much more mixing among batches, as the batch effects will have hopefully been corrected.
```

Comment:
We'll probably want to expand this, or maybe reduce it to a discussion question?